src/main/java/Parser.java

```diff
@@ -0,0 +1,80 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
+
+class Parser {
+    private static HashSet<String> keywords;
+    private static SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yyyy HHmm");
+
+    static {
+        keywords = new HashSet<>();
+        keywords.add("bye");
+        keywords.add("done");
+        keywords.add("todo");
+        keywords.add("event");
+        keywords.add("deadline");
+        keywords.add("list");
+        keywords.add("delete");
+        keywords.add("find");
+    }
+
+    static Command parse(String[] args) throws DukeException {
```

Comment:
Would it be better if instead of the HashSet keywords, this switch case is expanded so that for the todo, events, and deadline cases they call AddCommand()? Personally, I feel that it would make more sense to throw the DukeException in the default case, which will be reached if it fails through all the other switch cases that are the "defined commands"